core/primitives-core/src/profile.rs

```diff
@@ -91,41 +94,33 @@ impl ProfileData {
     /// with the help on the VM side, so we don't want to have profiling logic
     /// there both for simplicity and efficiency reasons.
     pub fn compute_wasm_instruction_cost(&mut self, total_gas_burnt: u64) {
-        let mut value = total_gas_burnt;
-        for cost in Cost::ALL {
-            value = value.saturating_sub(self[*cost]);
-        }
-        self[Cost::WasmInstruction] = value
+        self[Cost::WasmInstruction] =
+            total_gas_burnt.saturating_sub(self.action_gas()).saturating_sub(self.host_gas());
     }
 
-    pub fn get_action_cost(&self, action: ActionCosts) -> u64 {
+    fn get_action_cost(&self, action: ActionCosts) -> u64 {
         self[Cost::ActionCost { action_cost_kind: action }]
     }
 
     pub fn get_ext_cost(&self, ext: ExtCosts) -> u64 {
         self[Cost::ExtCost { ext_cost_kind: ext }]
     }
 
-    pub fn host_gas(&self) -> u64 {
-        let mut host_gas = 0u64;
-        for cost in Cost::ALL {
-            match cost {
-                Cost::ExtCost { ext_cost_kind: e } => host_gas += self.get_ext_cost(*e),
-                _ => {}
-            }
-        }
-        host_gas
+    fn host_gas(&self) -> u64 {
+        ExtCosts::iter().map(|a| self.get_ext_cost(a)).sum()
```

Comment:
While wee are here, let's maybe saturating_add ? Seem better to not think about overflows in profiles at all

Reply:
good call! changed it `fold` with saturating addition in the two places I used `sum` before